test/type/record.stmt

```diff
@@ -0,0 +1,184 @@
+40 columns                              |
+>>> Empty record.
+(   )  x;
+<<<
+() x;
+>>> Empty nullable record type.
+(  )  ?   x;
+<<<
+()? x;
+>>> Nullable record type.
+(  int  ,   bool  )   ?  x;
+<<<
+(int, bool)? x;
+>>> Single positional field.
+(  int  , ) x;
+<<<
+(int,) x;
+>>> Single named field.
+( { int  n  }  )  x;
+<<<
+({int n}) x;
+>>> Named positional fields.
+( int  value  ,  String  label) x;
+<<<
+(int value, String label) x;
+>>> Unnamed positional fields.
+( int    ,  String   ) x;
+<<<
+(int, String) x;
+>>> Named fields.
+(  {  int  value  ,  String  label  } ) x;
+<<<
+({int value, String label}) x;
+>>> Split named positional fields.
+( int  longValue  ,  String  veryVeryLongLabel  , ) x;
+<<<
+(
+  int longValue,
+  String veryVeryLongLabel,
+) x;
+>>> Unsplit unnamed positional fields have no trailing comma.
+( int    ,  String   ,  ) x;
+<<<
+(int, String) x;
+>>> Split only named fields.
+(  {  int  longValue  ,  String  veryLongLabel  ,  } ) x;
+<<<
+({
+  int longValue,
+  String veryLongLabel,
+}) x;
+>>> Empty record types don't split.
+someLongFunctionName__________________(() x) {}
+<<<
+someLongFunctionName__________________(
+  () x,
+) {}
+>>> Unsplit short single positional field.
+(TypeName,
+) x;
+<<<
+(TypeName,) x;
+>>> Unsplit single positional field.
+function((VeryLongTypeName____________,) x) {;}
```

Comment:
Unless you specifically think it's helpful to test these in parameter position, you can simplify them by just using a variable declaration instead of a function:

```dart
(VeryLongTypeName____________,) x;
```